src/modules/cpu/kernel/vignette.hpp

```diff
@@ -0,0 +1,1377 @@
+
+RppStatus vignette_u8_u8_host_tensor(Rpp8u *srcPtr,
+                                     RpptDescPtr srcDescPtr,
+                                     Rpp8u *dstPtr,
+                                     RpptDescPtr dstDescPtr,
+                                     Rpp32f *vignetteIntensityTensor,
+                                     RpptROIPtr roiTensorPtrSrc,
+                                     RpptRoiType roiType,
+                                     RppLayoutParams layoutParams)
+{
+    RpptROI roiDefault = {0, 0, (Rpp32s)srcDescPtr->w, (Rpp32s)srcDescPtr->h};
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(dstDescPtr->n)
+    for(int batchCount = 0; batchCount < dstDescPtr->n; batchCount++)
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f intensity = vignetteIntensityTensor[batchCount];
+        Rpp32s halfHeight = (Rpp32s) (roi.xywhROI.roiHeight >> 1);
+        Rpp32s halfWidth = (Rpp32s) (roi.xywhROI.roiWidth >> 1);
+        Rpp32f radius = std::max(halfHeight, halfWidth);
+        Rpp32f multiplier = -(0.25f * intensity) / (radius * radius);
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth;
+        Rpp32u alignedLength = bufferLength & ~15;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        __m256 pMultiplier = _mm256_set1_ps(multiplier);
+        __m256 pHalfWidth = _mm256_set1_ps(halfWidth);
+
+        // Vignette with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *srcPtrRow, *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            srcPtrRow = srcPtrChannel;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *srcPtrTemp, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                srcPtrTemp = srcPtrRow;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                Rpp32s iLoc = i - halfHeight;
+                Rpp32f iLocComponent = iLoc * iLoc;
+                __m256 pILocComponent = _mm256_set1_ps(iLocComponent);
+                __m256 pJLocComponent = _mm256_sub_ps(_mm256_setr_ps(0, 1, 2, 3, 4, 5, 6, 7), pHalfWidth);
```

Comment:
Please declare a register before the height loop and set it once
(_mm256_setr_ps(0, 1, 2, 3, 4, 5, 6, 7)) and use this register when needed, instead of creating new register every time

Please check if this register is already declared in any one of rpp_cpu_simd.hpp / rpp_cpu_common.hpp

Reply:
Done